Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -171,6 +193,72 @@ public class PushDeploymentController : ApiController
             return response;
         }
 
+        private async Task LocalZipHandler(IRepository repository, DeploymentInfoBase deploymentInfo, string targetBranch, ILogger logger, ITracer tracer)
+        {
+            if (_settings.RunFromLocalZip() && deploymentInfo is ZipDeploymentInfo zipDeploymentInfo)
+            {
+                // If this is a Run-From-Zip deployment, then we need to extract function.json
+                // from the zip file into path zipDeploymentInfo.SyncFunctionsTrigersPath
+                ExtractTriggers(repository, zipDeploymentInfo);
+            }
+            else
+            {
+                await LocalZipFetch(repository, deploymentInfo, targetBranch, logger, tracer);
+            }
+        }
+
+        private void ExtractTriggers(IRepository repository, ZipDeploymentInfo zipDeploymentInfo)
+        {
+            FileSystemHelpers.EnsureDirectory(zipDeploymentInfo.SyncFunctionsTrigersPath);
+            // Loading the zip file depends on how fast the file system is.
+            // Tested Azure Files share with a zip containing 120k files (160 MBs)
+            // takes 20 seconds to load. On my machine it takes 900 msec.
+            using (var zip = ZipFile.OpenRead(Path.Combine(_environment.SitePackagesPath, zipDeploymentInfo.ZipName)))
+            {
+                var entries = zip.Entries
+                    // Only select host.json, proxies.json, or top level directories.
+                    // Tested with a zip containing 120k files, and this took 90 msec
+                    // on my machine.
+                    .Where(e => e.FullName.Equals(Constants.FunctionsHostConfigFile, StringComparison.OrdinalIgnoreCase) ||
+                                e.FullName.Equals(Constants.ProxyConfigFile, StringComparison.OrdinalIgnoreCase) ||
```

Comment:
Shouldn't there be a test for host.json as well?

Reply:
as mentioned, the line above is the one for `host.json` (`Constants.FunctionsHostConfigFile == "host.json"`)